lms/djangoapps/grades/models.py

```diff
@@ -0,0 +1,42 @@
+from django.contrib.auth.models import User
+from django.db import models
+
+from xmodule_django.models import BlockTypeKeyField, CourseKeyField, UsageKeyField
+
+class Grade(models.Model):
+    """
+    A django model tracking persistent grades at the subsection level.
+    """
+
+    class Meta(object):
+        index_together = [
+            ['user_id', 'course_id', 'content_type', 'is_valid'],
+            ['course_id', 'updated'],
+            ['updated'],
+        ]
+
+    # TODO: timestamp discussions - do we need all three of [created, updated, subtree_edited_date]?
+    created = models.DateTimeField('creation timestamp', auto_now_add=True)
+    updated = models.DateTimeField('latest updated timestamp', auto_now=True)
+    subtree_edited_date = models.DateTimeField('last content edit timestamp')
+
+    # TODO: is this the right way to store course edit version? make sure it matches up w/ what Nimisha's exposing in her PR
+    course_version = models.CharField('guid of latest course version', max_length=255)
+
+    user = models.ForeignKey(User)
+    course_id = CourseKeyField(max_length=255)
+    content_id = UsageKeyField(max_length=255)
+    content_type = models.CharField('used to bucket grades by item type', max_length=255)
+    total_weighted_raw_score = models.IntegerField()
```

Comment:
Do we have a firm line on ints vs. floats here (since CSM uses floats)? I'm fine with the decision to use ints, as long as it's been thought through.

Also, what is an empty sequence (something marked as a HW but with no problems in it)? 0/0?